cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaMetadataStateManagerMockTest.java

```diff
@@ -0,0 +1,640 @@
+                .endJbodStorage()
+                .withRoles(ProcessRoles.BROKER)
+                .withResources(new ResourceRequirementsBuilder().withRequests(Map.of("cpu", new Quantity("6"))).build())
+            .endSpec()
+            .build();
+    private static final List<KafkaPool> POOLS = NodePoolUtils.createKafkaPools(Reconciliation.DUMMY_RECONCILIATION, KAFKA, List.of(CONTROLLERS, BROKERS), Map.of(), Map.of(), true, SHARED_ENV_PROVIDER);
+
+    /** Tests migration from Zookeeper State to KRaftPostMigration State.
+     *  The strimzi.io/kraft annotation is set to `migration`
+     *
+     * @param context Test context
+     */
+    @Test
+    public void testMigrationFromZooKeeperToKRaftPostMigrationState(VertxTestContext context) {
+
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+
+        SecretOperator mockSecretOps = supplier.secretOperations;
+        Secret secret = new Secret();
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(KafkaResources.clusterCaCertificateSecretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(KafkaResources.secretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+
+        MockKafkaReconciler kr = new MockKafkaReconciler(
+                new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME),
+                vertx,
+                CONFIG,
+                supplier,
+                new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+                KAFKA,
+                List.of(BROKERS, CONTROLLERS),
+                createKafkaCluster(KAFKA),
+                CLUSTER_CA,
+                CLIENTS_CA);
+
+        KafkaStatus status = new KafkaStatus();
+
+        Checkpoint async = context.checkpoint();
+        kr.reconcile(status, Clock.systemUTC()).onComplete(res -> context.verify(() -> {
+            assertThat(res.succeeded(), is(true));
+            assertEquals(status.getKafkaMetadataState(), KRaftMigration.name());
+        })).compose(v -> kr.reconcile(status, Clock.systemUTC())).onComplete(context.succeeding(v -> {
+            assertEquals(status.getKafkaMetadataState(), KRaftDualWriting.name());
+        })).compose(v -> kr.reconcile(status, Clock.systemUTC())).onComplete(context.succeeding(v -> {
+            assertEquals(status.getKafkaMetadataState(), KRaftPostMigration.name());
+            async.flag();
+        }));
+    }
+
+    /** Tests migration from KRaftPostMigration State to KRaft State.
+     *  The strimzi.io/kraft annotation is set to `enabled`
+     *
+     * @param context Test Context
+     */
+    @Test
+    public void testKRaftPostMigrationToKRaftState(VertxTestContext context) {
+
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editMetadata()
+                    .addToAnnotations(Annotations.ANNO_STRIMZI_IO_KRAFT, "enabled")
```

Comment:
this is kind of what you should do in the previous test but with the "migration" value :-)